imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js

```diff
@@ -0,0 +1,367 @@
+      newTag: {
+        name: ""
+      }
+    };
+  }
+
+  componentDidMount() {
+    $(window).on("resize", this.onWindowResize).trigger("resize");
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { tagIds, tagsByKey } = nextProps;
+    this.setState({ tagIds, tagsByKey });
+  }
+
+  componentWillUnmount() {
+    $(window).off("resize", this.onWindowResize);
+  }
+
+  onWindowResize = () => {
+    if (window.matchMedia("(max-width: 991px)").matches) {
+      this.setState({
+        [NavbarStates.Orientation]: NavbarOrientation.Vertical,
+        [NavbarStates.Position]: NavbarPosition.Fixed,
+        [NavbarStates.Anchor]: NavbarAnchor.Left
+      });
+    } else {
+      this.setState({
+        [NavbarStates.Orientation]: NavbarOrientation.Horizontal,
+        [NavbarStates.Position]: NavbarPosition.Static,
+        [NavbarStates.Anchor]: NavbarAnchor.None,
+        [NavbarStates.Visible]: false
+      });
+    }
+  }
+
+  handleNewTagSave = (tag, parentTag) => {
+    editable = true; // keep editing state after re-render
+    TagNavHelpers.onTagCreate(tag.name, parentTag);
+    this.setState({ newTag: { name: "" } });
+  }
+
+  handleNewTagUpdate = (tag) => { // updates the current tag state being edited
+    this.setState({
+      newTag: tag
+    });
+  }
+
+  handleTagRemove = (tag, parentTag) => {
+    editable = true; // keep editing state after re-render
+    TagNavHelpers.onTagRemove(tag, parentTag);
+  }
+
+  handleTagUpdate = (tag) => {
+    const newState = update(this.state, {
+      tagsByKey: {
+        [tag._id]: {
+          $set: tag
+        }
+      }
+    });
+
+    this.setState(newState);
+  }
+
+  handleTagSave = (tag) => {
+    TagNavHelpers.onUpdateTag(tag._id, tag.name);
+  }
+
+  handleMoveTag = (dragIndex, hoverIndex) => {
+    const tag = this.state.tagIds[dragIndex];
+
+    // Apply new sort order to variant list
+    const newState = update(this.state, {
+      tagIds: {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, tag]
+        ]
+      }
+    });
+
+    // Set local state so the component does't have to wait for a round-trip
+    // to the server to get the updated list of variants
+    this.setState(newState, () => {
+      editable = true; // keep editing state after re-render
+      debounce(() => TagNavHelpers.onTagSort(this.state.tagIds), 500)(); // Save the updated positions
+    });
+  }
+
+  handleGetSuggestions = (suggestionUpdateRequest) => {
+    const suggestions = TagNavHelpers.updateSuggestions(
+      suggestionUpdateRequest.value,
+      { excludeTags: this.state.tagIds }
+    );
+
+    this.setState({ suggestions });
+  }
+
+  handleClearSuggestions = () => {
+    this.setState({ suggestions: [] });
+  }
+
+  get canEdit() {
+    return this.props.hasEditRights && Reaction.isPreview() === false;
+  }
+
+  attachBodyListener = () => {
+    document.body.addEventListener("mouseover", this.closeDropdown);
```

Comment:
jQuery in React. ouch..
https://facebook.github.io/react/docs/events.html#mouse-events